src/border_agent/border_agent.cpp

```diff
@@ -403,33 +377,44 @@ void BorderAgent::HandleMdnsState(Mdns::Publisher::State aState)
     return;
 }
 
-static uint64_t ConvertTimestampToUint64(const otTimestamp &aTimestamp)
+void BorderAgent::HandleBorderAgentStateChange(bool aIsActive, uint16_t aPort)
 {
-    // 64 bits Timestamp fields layout
-    //-----48 bits------//-----15 bits-----//-------1 bit-------//
-    //     Seconds      //      Ticks      //  Authoritative    //
-    return (aTimestamp.mSeconds << 16) | static_cast<uint64_t>(aTimestamp.mTicks << 1) |
-           static_cast<uint64_t>(aTimestamp.mAuthoritative);
+    VerifyOrExit(aIsActive != mBaIsActive || aPort != mMeshCoPUdpPort);
+
+    mBaIsActive     = aIsActive;
+    mMeshCoPUdpPort = aPort;
+
+    UpdateMeshCoPService();
+
+exit:
+    return;
 }
 
-#if OTBR_ENABLE_BORDER_ROUTING
-void AppendOmrTxtEntry(otInstance &aInstance, Mdns::Publisher::TxtList &aTxtList)
+void BorderAgent::HandleOtMeshCoPTxtValueChange(const std::vector<uint8_t> &aOtMeshCoPTxtValues)
 {
-    otIp6Prefix       omrPrefix;
-    otRoutePreference preference;
+    mOtMeshCoPTxtValues.assign(aOtMeshCoPTxtValues.begin(), aOtMeshCoPTxtValues.end());
 
-    if (OT_ERROR_NONE == otBorderRoutingGetFavoredOmrPrefix(&aInstance, &omrPrefix, &preference))
+    // Parse extended address from the encoded data for the first time
+    if (!mIsInitialized)
     {
-        std::vector<uint8_t> omrData;
+        Mdns::Publisher::TxtList txtList;
 
-        omrData.reserve(1 + OT_IP6_PREFIX_SIZE);
-        omrData.push_back(omrPrefix.mLength);
-        std::copy(omrPrefix.mPrefix.mFields.m8, omrPrefix.mPrefix.mFields.m8 + (omrPrefix.mLength + 7) / 8,
-                  std::back_inserter(omrData));
-        aTxtList.emplace_back("omr", omrData.data(), omrData.size());
+        Mdns::Publisher::DecodeTxtData(txtList, mOtMeshCoPTxtValues.data(), mOtMeshCoPTxtValues.size());
+        for (auto &entry : txtList)
+        {
+            if (entry.mKey == "xa")
+            {
+                memcpy(mExtAddress.m8, entry.mValue.data(), sizeof(otExtAddress));
```

Comment:
```suggestion
                memcpy(mExtAddress.m8, entry.mValue.data(), sizeof(mExtAddress.m8));
```
Not sure if this will be more readable. Feel free to share your thoughts :).

Reply:
Agree. Using the same thing in a line is better.